drivers/spi/spi_ll_stm32.c

```diff
@@ -97,13 +96,13 @@ static void spi_stm32_shift_m(SPI_TypeDef *spi, struct spi_stm32_data *data)
 
 	if (SPI_WORD_SIZE_GET(data->ctx.config->operation) == 8) {
 		rx_frame = LL_SPI_ReceiveData8(spi);
-		if (spi_context_rx_on(&data->ctx)) {
+		if (spi_context_rx_buf_on(&data->ctx)) {
 			UNALIGNED_PUT(rx_frame, (u8_t *)data->ctx.rx_buf);
 			spi_context_update_rx(&data->ctx, 1, 1);
```

Comment:
it's not working if `spi_context_update_rx` is not called (hanging forever when rx buffer is NULL). 
compared this revision with the two versions below:

working (this PR up until about 1 hour ago, but without tbursztyka's API-changes) : 
```c
if (SPI_WORD_SIZE_GET(data->ctx.config->operation) == 8) {
	rx_frame = LL_SPI_ReceiveData8(spi);
	if (data->ctx.rx_buf) {
		UNALIGNED_PUT(rx_frame,
			      (u8_t *)data->ctx.rx_buf);
        }
	spi_context_update_rx(&data->ctx, 1, 1);
} else {
```
also working (my solution combining both versions above) : 
```c
if (SPI_WORD_SIZE_GET(data->ctx.config->operation) == 8) {
	rx_frame = LL_SPI_ReceiveData8(spi);
	if (spi_context_rx_buf_on(&data->ctx)) {
		UNALIGNED_PUT(rx_frame,
			      (u8_t *)data->ctx.rx_buf);
        }
	spi_context_update_rx(&data->ctx, 1, 1);
} else {
```

Reply:
You are right, I messed it up during a rebase.